src/Solr/Query/AbstractParamsBuilder.php

```diff
@@ -148,9 +148,15 @@ public function getStoreId()
      * @param string $attributeToReset
      * @return string
      */
-    private function getFilterQuery($attributeToReset = '')
+    protected function getFilterQuery($attributeToReset = '')
     {
-        return $this->filterQueryBuilder->buildFilterQuery($this->getStoreId(), $attributeToReset);
+        $filterQuery = $this->filterQueryBuilder->buildFilterQuery($this->getStoreId(), $attributeToReset);
+
+        if (!$this->resultsConfig->isShowOutOfStock()) {
+            $filterQuery .= ' AND -is_in_stock_i:0';
+        }
```

Comment:
any chance to move this into the filter query builder instead of duplicating it in several params builder implementations? the responsibilities are not strictly separated anymore